lib/oauth2/response.rb

```diff
@@ -114,6 +123,15 @@ def parser
 
       @parser ||= @@parsers[@@content_types[content_type]]
     end
+
+    private
+
+    def to_snake_case(string)
+      string.gsub(/([A-Z]+)([A-Z][a-z])/, '\1_\2')
+            .gsub(/([a-z\d])([A-Z])/, '\1_\2')
+            .tr("-", "_")
+            .downcase
```

Comment:
The snake casing in [`plissken `](https://github.com/futurechimp/plissken/blob/master/lib/plissken/methods.rb) is a bit more robust.
```
        string.tr("::", "/")
              .gsub(/([A-Z]+)([A-Z][a-z])/, '\1_\2')
              .gsub(/([a-z\d])([A-Z])/, '\1_\2')
              .tr("-", "_")
              .downcase
```
Let's switch to **plissken** and remove this internal snake case logic.  We'll get much more robust testing as a result.

Reply:
Thanks. Agree that `plissken` changes are to widely. 
I think this code, which from #202, is original from `plissken`. I remove `tr("::", "/")` because I think it's mostly use to transform full class name to snake case. In our case, I believe we don't have `::` in header name.